tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -52,29 +53,30 @@ public TracerSettings(bool useDefaultSources)
         /// using the specified <see cref="IConfigurationSource"/> to initialize values.
         /// </summary>
         /// <param name="source">The <see cref="IConfigurationSource"/> to use when retrieving configuration values.</param>
-        public TracerSettings(IConfigurationSource source)
+        public TracerSettings(IConfigurationSource? source)
         {
             var commaSeparator = new[] { ',' };
+            source ??= NullConfigurationSource.Instance;
```

Comment:
I had checked if other Settings should use this line as well. But I had missed the `SecuritySettings` that are not in that folder. Maybe you should do the same there?

Reply:
Oh yeah, I'll look at annotating those too 👍